### PR TITLE
Feature diann anomaly

### PR DESCRIPTION
**Pull request by devonjkohler**
# Motivation and Context

Updated DIA-NN converter to work with MSstats+ and some minor bug fixes to anomaly detection model.

## Changes

- Added quality metric columns to DIA-NN converter
- Made sure anomaly scores were calculated on unique rows
- Removed clipping because MSstats will reweight anomaly scores in dataProcess now  

## Testing

Please describe any unit tests you added or modified to verify your changes.

## Checklist Before Requesting a Review
- [x] I have read the MSstats [contributing guidelines](https://github.com/Vitek-Lab/MSstatsConvert/blob/master/.github/CONTRIBUTING.md)
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

## Motivation and Context

This PR updates the DIA-NN -> MSstats conversion pipeline to integrate anomaly-detection support for MSstats+ and fixes bugs in the anomaly scoring pipeline. Goals: add quality-metric columns to the DIANN converter, ensure anomaly scores are computed on unique (aggregated) rows, and remove clipping of anomaly scores so MSstats' dataProcess can reweight scores.

## Short solution summary

- Add anomaly-related parameters and propagate anomaly metric columns through cleaning, preprocessing, and balanced-design steps.
- Compute anomaly scores on unique model-input rows and merge scores back to original rows to preserve uniqueness.
- Replace prior CuSum-like builders with stateful, NA-aware implementations and remove clipping of final anomaly scores.
- Ensure anomaly metrics are included in outputs only when anomaly scoring is enabled.

## Detailed changes

- R/converters_DIANNtoMSstatsFormat.R
  - Signature extended with new anomaly parameters: calculateAnomalyScores (FALSE), anomalyModelFeatures (c()), anomalyModelFeatureTemporal (c()), removeMissingFeatures (0.5), anomalyModelFeatureCount (100), runOrder (NULL), n_trees (100), max_depth ("auto"), numberOfCores (1), plus propagation of these through the pipeline.
  - Standardize anomalyModelFeatures early (.standardizeColnames) and pass anomaly_metrics into MSstatsClean, MSstatsPreprocess, MSstatsBalancedDesign.
  - Change filtering behavior: entries failing qvalue/PGQ thresholds are zeroed (Intensity <- 0 or NA adjustments) rather than subset-removed.
  - Conditional call to MSstatsAnomalyScores when calculateAnomalyScores = TRUE.
  - Minor leftover debugging comment: browser() call(s) remain (see coding guideline note).

- R/clean_DIANN.R
  - .cleanRawDIANN() and .cleanDIANNSelectRequiredColumns() accept calculateAnomalyScores and anomalyModelFeatures.
  - When anomaly scoring enabled, anomalyModelFeatures appended to required columns (qual_cols) so quality metric columns are retained.

- R/utils_anomaly_score.R
  - Reimplemented mean_increase, mean_decrease, dispersion_increase as stateful, NA-aware builders with proper initialization and propagation of NA.
  - .prepareSpectronautAnomalyInput now builds unique model_input (unique combinations of split column and quality metrics) for anomaly-model training.
  - .runAnomalyModel computes scores on model_input, assigns AnomalyScores there, then merges scores back into input_data by split_column + quality metrics to ensure unique-row scoring.
  - Removed clipping of anomaly scores in this flow (scores are left as produced by the forest).

- R/MSstatsConvert_core_functions.R
  - Expanded Roxygen/documentation for MSstatsAnomalyScores and related parameters; documentation only, no signature changes.

- man/*.Rd updates
  - man/DIANNtoMSstatsFormat.Rd: updated to document the new anomaly-related public parameters and default values.
  - man/MSstatsAnomalyScores.Rd: expanded description of the isolation-forest approach, missing-data handling (median imputation fallback), and parallel support.
  - man/MSstatsClean.Rd and man/dot-cleanRawDIANN.Rd: documented the new arguments for DIANN cleaning.

- inst/tinytest/test_utils_anomaly_score.R
  - Added comprehensive tests (369 lines) covering property-based checks, extreme-value/outlier detection, reproducibility/consistency, edge cases (identical/minimal/all-zero), scale invariance, regression properties (non-negative/finite), builders (mean_increase/mean_decrease/dispersion_increase), and temporal/missing-run parameters.

- src/isolation_forest.cpp
  - Large block of commented-out alternative/previous Isolation Forest implementation and notes added; active interface/behavior unchanged.

- Repository metadata
  - .gitignore: adjusted patterns (re-added dll ignore pattern and added .lintr and .vscode).
  - DESCRIPTION: RoxygenNote bumped 7.3.2 -> 7.3.3.

## Unit tests added or modified

- inst/tinytest/test_utils_anomaly_score.R (new/expanded)
  - Tests include:
    - Monotonicity/property tests (progressive CuSum-like behavior).
    - Outlier detection checks (obvious outliers get highest scores).
    - Reproducibility/correlation checks across runs (>0.95 expected).
    - Top-k overlap checks (top-5 anomalies overlap).
    - Edge cases: identical values, minimal datasets, all-zero data.
    - Scale invariance (relative rankings preserved after scaling).
    - Builder function unit tests for mean_increase, mean_decrease, dispersion_increase with expected numeric outputs.
    - Tests for .prepareSpectronautAnomalyInput temporal/feature-count/missing-run behavior.
  - Existing DIANN converter/cleaning tests remain focused on structure/required columns.

## Coding guidelines / issues identified

- Debugging artifacts: one or more browser() calls remain commented/uncommented in R/converters_DIANNtoMSstatsFormat.R (observed at least one "browser()" near end of file). These should be removed before merging.
- Large commented blocks in src/isolation_forest.cpp increase file size and may reduce readability; consider moving historical code to a separate notes file if retention is required.
- No public API signature removals; several public signatures expanded—ensure downstream callers are aware of new parameters and defaults.

### CODE REVIEW

File: R/converters_DIANNtoMSstatsFormat.R [hunk 1]
```diff
@@ -52,26 +61,36 @@
...
+
     input = MSstatsConvert::MSstatsImport(list(input = input),
                                           "MSstats", "DIANN")
-    input = MSstatsConvert::MSstatsClean(input, MBR = MBR, 
-                                         quantificationColumn = quantificationColumn)
+    # browser()
+    input = MSstatsConvert::MSstatsClean(
+        input, MBR = MBR, quantificationColumn = quantificationColumn,
+        calculateAnomalyScores = calculateAnomalyScores, 
+        anomalyModelFeatures = anomalyModelFeatures)
```
[the remark this answers is not in the document]
Comment (devonjkohler) [reply R/converters_DIANNtoMSstatsFormat.R]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Vitek-Lab/MSstatsConvert/pull/119#discussion_r2872989039)

File: R/converters_DIANNtoMSstatsFormat.R [hunk 2]
```diff
@@ -52,26 +61,36 @@
...
+    removeProtein_with1Feature = TRUE,
+    MBR = TRUE, 
+    quantificationColumn = "FragmentQuantCorrected",
+    calculateAnomalyScores=FALSE, anomalyModelFeatures=c(),
+    anomalyModelFeatureTemporal=c(), removeMissingFeatures=.5,
+    anomalyModelFeatureCount=100,
+    runOrder=NULL, n_trees=100, max_depth="auto", numberOfCores=1, 
+    use_log_file = TRUE, append = FALSE, 
+    verbose = TRUE, log_file_path = NULL,
+    ...) {
```
[the remark this answers is not in the document]
Comment (devonjkohler) [reply R/converters_DIANNtoMSstatsFormat.R:80]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Vitek-Lab/MSstatsConvert/pull/119#discussion_r2872989065)
